src/gui/newwizard/pages/ServerUrlSetupWizardPage.cpp

```diff
@@ -0,0 +1,31 @@
+#include "ServerUrlSetupWizardPage.h"
+
+#include "theme.h"
+#include "ui_ServerUrlSetupWizardPage.h"
+
+namespace OCC::Wizard {
+
+ServerUrlSetupWizardPage::ServerUrlSetupWizardPage(const QUrl &serverUrl)
+    : _ui(new ::Ui::ServerUrlSetupWizardPage)
+{
+    _ui->setupUi(this);
+
+    _ui->welcomeTextLabel->setText(tr("Welcome to %1").arg(Theme::instance()->appNameGUI()));
+
+    _ui->urlLineEdit->setText(serverUrl.toString());
+
+    // first, we declare this object as an event filter
+    // this then allows us to make the controller the event filter of the current page
+    _ui->urlLineEdit->installEventFilter(this);
```

Comment:
Shouldn't this be handled by the event filter in the window?

Reply:
In the current state the signal doesn't seem to be propagated to the window. But I've changed a lot of things since. I'll try.

Reply:
Found it. Have to install the window as its own event filter. Seems a bit awkward at first, but actually makes sense.